### ISSUE TITLE
Allow in-place un-setting of WCS.wcs.aux

### ISSUE THREAD

**Issue by dstansby** [astropy/astropy#11041]
https://github.com/astropy/astropy/pull/10333 added support for auxiliary parameters on a WCS object, accessed by WCS.wcs.aux. However, once set these are currently immutable. It would be nice if they could be modified in place, which we have a use case for in sunpy. At the moment we have to convert a pre-existing WCS to a header, modify the aux fits keywords, and then create anew WCS.

**Comment by mcara** [comment]
I'll try to work on this at the end of the week.

**Comment by mcara** [comment]
Sorry for the delay. Could you please elaborate on exactly how would you like things to work: I am likely missing something but it seems that setting `aux` attributes is already supported. Could you, please, provide an example that shows what is currently not supported but you would like it to work?

**Comment by Cadair** [comment]
I am not sure about the individual attributes, but you can't clear the whole struct I don't think?

**Comment by mcara** [comment]
What do you mean?

> ... support for auxiliary parameters on a WCS object, accessed by WCS.wcs.aux. However, once set _these_ [auxiliary parameters] are currently immutable. It would be nice if they could be modified in place ...

It looks like @dstansby wanted to be able to modify individual parameters of the `aux` structure.

> ...but you can't clear the whole struct

What do you mean? Could you, please, provide an example?

**Comment by dstansby** [comment]
Sorry for the complete lack of example... I just tried to write one, and setting them in place seemed to work fine. Let me double check it does what we want it to do in sunpy, and I'll then either close this or post a self contained example.

**Comment by dstansby** [comment]
Ah, so the issue is we want to "un-set" some of the metadata values after they have been first set, e.g. do this
```python
from astropy.io import fits
from astropy.wcs import WCS
from astropy.utils.data import get_pkg_data_filename
fn = get_pkg_data_filename('data/j94f05bgq_flt.fits', package='astropy.wcs.tests')
f = fits.open(fn)
w = WCS(f[1].header)

print(w.wcs.aux.rsun_ref)
w.wcs.aux.rsun_ref = 1
w.wcs.aux.rsun_ref = None
```
which currently raises
```python
Traceback (most recent call last):
  File "/Users/dstansby/github/astropy/test.py", line 10, in <module>
    w.wcs.aux.rsun_ref = None
TypeError: must be real number, not NoneType
```
At the moment we have to convert the WCS to a header, remove the un-wanted keywords from the header, and create a new WCS from the header. It would be great if we could "un-set" the values by assigning them in-place to `None`.

**Comment by mcara** [comment]
I got it. I will look into this. Thanks for clarification!

**Comment by mcara** [comment]
See fix in #11166

**Comment by dstansby** [comment]
Thanks a lot!
